### PR TITLE
Refactor protocol

### PR DESCRIPTION
**Pull request by mweichert**
# 🌮 Taqueria PR

## 🪁 Description

Protocol refactor merged with main.

## 🪂 Pre-Merge Checklist (Definition of Done)

🚦 Required to merge:

- [x] ⛱️ I have completed this PR template in full and updated the title appropriately
- [x] ⛵ My code builds cleanly, and I have manually tested the changes
- [ ] 🏄‍♂️ Another team member has built this branch and done manual testing on the change
- [x] 🏖️ New and existing unit tests pass locally and in CI
- [x] 🔱 The test plan has been implemented and verified by an SDET
- [x] 🦀 Automated tests have been written and added to this PR
- [x] 🐬 I have commented my code, particularly in hard-to-understand areas
- [x] 🤿 Corresponding changes have been made to all documentation
- [x] 🐚 Required changes to the VScE have been made
- [ ] 🪸 Required updates to scaffolds have been made
- [ ] 🚢 The release checklist has been completed

## 🛩️ Summary of Changes

In terms of scope of work, this is remaining:
- Consolidate `@taqueria/protocol` and `@taqueria/protocol-types` packages
- Our factory methods aren't formatting the Zod errors.

## 🎢 Test Plan

_Please describe the testing strategy and plan for this PR. Keep this lightweight and anticipate any testing challenges_

## 🛸 Type of Change

- [ ] 🧽 Chore ➾
- [ ] 🛠️ Fix ➾
- [ ] ✨ Feature ➾
- [ ] 👷 Refactor ➾
- [ ] 🧪 Pre-Release ➾
- [ ] 🚀 Release ➾

### CODE REVIEW

**Review by jchenche — CHANGES_REQUESTED** [review]
Can we merge main in and resolve any conflicts?

And why do we have 2 PRs for the same thing? https://github.com/ecadlabs/taqueria/pull/1471 and this

**Comment by mweichert** [comment]
> Can we merge main in and resolve any conflicts?

Yep, will do. I have to do this several times a day. 

> And why do we have 2 PRs for the same thing? #1471 and this

The old branch is mostly abandoned but around for reference till we get this merged into the pre-release branch.

This was mentioned in a standup a little while ago. The first attempt was trying to merge and consolidate all in one go, which wasn't working well. So we I created a new branch from an earlier commit of the `refactor-protocal` branch and focused just on merging main and getting the tests to pass. That's what this PR represents, although I'm going to work on the consolidation part now.

**Review by AlirezaHaghshenas — APPROVED** [review]

File: taqueria-types.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+debugger;
```
Comment (AlirezaHaghshenas) [inline taqueria-types.ts:1]: Do we want this be merged?

**Review by danielelisi — APPROVED** [review]
Changes to Github Workflows LGTM :+1:

File: .gitignore [hunk 2]
```diff
@@ -102,7 +102,7 @@ dist
 
 # TernJS port file
 .tern-port
-
+v
```
[the remark this answers is not in the document]
Comment (mweichert) [reply .gitignore]: Fixed!

File: .gitignore [hunk 3]
```diff
@@ -113,6 +113,8 @@ node_modules
 /taqueria
 /taq
 tests/coverage
+/taqueria-protocol/out
+/taqueria-protocol/out
```
[the remark this answers is not in the document]
Comment (mweichert) [reply .gitignore]: Fixed!

File: taqueria-types.ts (hunk 1 above)
Comment (mweichert) [reply taqueria-types.ts:1]: Good find. I've removed that one, as well as a couple of others!. Thanks Ali!

**Pull request merged**